lib/charms/alertmanager_k8s/v0/alertmanager_remote_configurer.py

```diff
@@ -0,0 +1,1940 @@
+
+        Args:
+            event: Juju RelationJoinedEvent
+
+        Returns:
+            None
+        """
+        self.load_initial_alertmanager_config_to_relation_data_bag(event)
+
+    def load_initial_alertmanager_config_to_relation_data_bag(
+        self, event: RelationJoinedEvent
+    ) -> None:
+        """Reads current Alertmanager configuration from alertmanager.yml and puts the content
+        inside relation data bag. If needed, it can be used by `alertmanager-remote-config-write`
+        consumer as a starting point for generating custom configuration.
+
+        Args:
+            event: Juju RelationJoinedEvent
+
+        Returns:
+            None
+        """
+        if not self._charm.unit.is_leader():
+            return
+        alertmanager_config = self._get_current_config()
+        event.relation.data[self._charm.app]["alertmanager_config"] = json.dumps(
+            alertmanager_config
+        )
+
+    def config(self) -> dict:
+        config = {}
+        remote_config_write_relation = self._charm.model.get_relation(self._relation_name)
+        if remote_config_write_relation:
+            try:
+                raw_config = remote_config_write_relation.data[remote_config_write_relation.app][  # type: ignore[index]  # noqa: E501
+                    "alertmanager_config"
+                ]
+                if self._config_is_valid(json.loads(raw_config)):
+                    config = raw_config
+            except KeyError:
+                logger.warning(
+                    "Remote config provider relation exists, but no config has been provided."
+                )
+        return config
+
+    def templates(self) -> list:
+        templates = []
+        remote_config_write_relation = self._charm.model.get_relation(self._relation_name)
+        if remote_config_write_relation:
+            try:
+                templates_raw = remote_config_write_relation.data[
+                    remote_config_write_relation.app  # type: ignore[index]
+                ]["alertmanager_templates"]
+                templates = json.loads(templates_raw)
+            except KeyError:
+                logger.warning(
+                    "Remote config provider relation exists, but no templates have been provided."
+                )
+        return templates
+
+    @staticmethod
+    def _config_is_valid(config: dict) -> bool:
+        try:
+            validate(instance=config, schema=ALERTMANAGER_CONFIG_JSON_SCHEMA)
+            return True
+        except exceptions.ValidationError as e:
+            logger.error(f"Config validation error: {e.message}")
+            return False
+
+    def _get_current_config(self):
+        config_endpoint = "api/v2/status"
+        url = f"{self.api_address}/{config_endpoint}"
+        response = requests.get(url)
+        return response.json()["config"]["original"] if response.status_code == 200 else ""
+
+
+class AlertmanagerRemoteConfigurerConsumer(Object):
```

Comment:
This would probably make more sense. Is there any official guideline for that? 
`prometheus-remote-write` library (which this lib was roughly based on), defines the `Provider` and the `Consumer`...